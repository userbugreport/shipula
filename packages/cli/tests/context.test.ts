import path from "path";
import {
  buildDeployProps,
  buildInfoProps,
  buildEnvProps,
  getInternalPath,
  getStackPath,
} from "../src/context";

describe("Context", () => {
  const sampleIn = path.join(__dirname, "..", "..", "express-sample");
  it("can build deploy props", async () => {
    const p = await buildDeployProps(sampleIn);
    expect(p).toMatchSnapshot();
  });
  it("can build info props with default stack", async () => {
    const p = await buildInfoProps(sampleIn);
    expect(p).toMatchSnapshot();
  });
  it("can build info props with named stack", async () => {
    const p = await buildInfoProps(sampleIn, "dev");
    expect(p).toMatchSnapshot();
  });
  it("can build env props with named stack", async () => {
    const p = await buildEnvProps(sampleIn, "dev", ["a=b", "/junk@=stuff"]);
    expect(p).toMatchSnapshot();
  });
  it("can build env props with default stack", async () => {
    const p = await buildEnvProps(sampleIn, "a=b");
    expect(p).toMatchSnapshot();
  });
  it("can build a stack path", async () => {
    const p = await getStackPath("a", "b");
    expect(p).toEqual("shipula/a/b");
  });
  it("can build a system path", async () => {
    const p = await getInternalPath("a", "b");
    expect(p).toEqual(".shipula/a/b");
  });
});
