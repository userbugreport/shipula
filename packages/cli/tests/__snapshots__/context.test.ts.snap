// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context can build deploy props 1`] = `
Object {
  "package": Object {
    "dependencies": Object {
      "express": "^4.17.1",
    },
    "from": "/Users/wballard/github/userbugreport/shipula/packages/express-sample",
    "license": "MIT",
    "main": "index.js",
    "name": "@shipula/express-sample",
    "scripts": Object {
      "start": "node index.js",
    },
    "version": "0.0.0",
  },
  "stackName": "default",
}
`;

exports[`Context can build env props with default stack 1`] = `
Object {
  "package": Object {
    "name": "@shipula/express-sample",
  },
  "setVariables": Object {
    "a": "b",
  },
  "stackName": "default",
}
`;

exports[`Context can build env props with named stack 1`] = `
Object {
  "package": Object {
    "name": "@shipula/express-sample",
  },
  "setVariables": Object {
    "a": "b",
    "junk": "stuff",
  },
  "stackName": "dev",
}
`;

exports[`Context can build info props with default stack 1`] = `
Object {
  "package": Object {
    "name": "@shipula/express-sample",
  },
  "stackName": "default",
}
`;

exports[`Context can build info props with named stack 1`] = `
Object {
  "package": Object {
    "name": "@shipula/express-sample",
  },
  "stackName": "dev",
}
`;
